src/Microsoft.ML.OnnxTransform/OnnxUtils.cs

```diff
@@ -4,27 +4,26 @@
 
 using Microsoft.ML.Runtime;
 using Microsoft.ML.Runtime.Data;
-using Microsoft.ML.Runtime.Internal.Utilities;
-using Microsoft.ML.Scoring;
+using Microsoft.ML.OnnxRuntime;
+using System.Numerics.Tensors;
 using System;
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
-using Microsoft.ML.StaticPipe;
 
-using OnnxShape = System.Collections.Generic.List<long>;
-using Microsoft.ML.Data;
+using OnnxShape = System.Collections.Generic.List<int>;
 
 namespace Microsoft.ML.Transforms
 {
     /// <summary>
-    /// OnnxModel is a facad for ModelManager. ModelManager is provided by Sonoma API,
-    /// and it has a lot of functionality (multiple models, multiple versions) that are not
-    /// needed by Onnx transform, which only needs a single model. This facad simplifies the
-    /// usage of onnx model.
+    /// OnnxModel is a utility class to load ONNX models and retrieve metadata
+    /// for inputs and outputs. The metadata includes the names, shapes and types
+    /// It provides API to open a session, score tensors (NamedOnnxValues) and return
+    /// the results.
     /// </summary>
-    internal sealed class OnnxModel
+    public sealed class OnnxModel
```

Comment:
This should remain `internal` unless it absolutely cannot be. We are working hard at minimizing our public surface area. See #1519.

Reply:
If this is necessary for consumers to get model information, we should think hard whether `Microsoft.ML.OnnxTransform` is the right library for these APIs to exist, or if consumers should just use `Microsoft.ML.OnnxRuntime` itself. Do we really want to public APIs for getting info about Onnx models? Or should the `OnnxRuntime` be the right place for that?

---
In reply to: [236336770](https://github.com/dotnet/machinelearning/pull/1717#discussion_r236336770) [](ancestors = 236336770)

Reply:
That's a good point. I'll make this internal -- and use OnnxRuntime in the example to retrieve this metadata. As long as the functionality exists somewhere.  

---
In reply to: [236337918](https://github.com/dotnet/machinelearning/pull/1717#discussion_r236337918) [](ancestors = 236337918,236336770)